### PR TITLE
Prepare v0.24.0 release.

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Release v0.24.0.

## Description of Changes

Provided in commit messages.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: CREDITS.md [hunk 1]
```diff
@@ -38,11 +38,13 @@ Contributors to the Firecracker repository:
 * Andrii Radyk <ander.ender@gmail.com>
 * Angus McInnes <angus@amcinnes.info>
 * Arun Gupta <arun.gupta@gmail.com>
+* Arunav Sanyal <Khalian@users.noreply.github.com>
 * Atsushi Ishibashi <atsushi.ishibashi@finatext.com>
 * Aussie Schnore <aussiev123@yahoo.com>
 * Babis Chalios <babis.chalios@gmail.com>
 * Begley Brothers Inc <begleybrothers@gmail.com>
 * Benjamin Fry <benjaminfry@me.com>
+* berciuliviu <lberciu@amazon.com>
```
Comment (alindima) [inline CREDITS.md:47]: Should be formatted as `Liviu Berciu`. You could edit the `.mailmap` file

**Review by AlexandruCihodaru — APPROVED** [review]
🚀

File: CREDITS.md [hunk 2]
```diff
@@ -71,17 +73,20 @@ Contributors to the Firecracker repository:
 * George Pisaltu <gpl@amazon.com>
 * german gomez <germangb42@gmail.com>
 * Greg Dunn <gregdunn@amazon.com>
+* Gulshan Kumar <kumargu@amazon.com>
```
Comment (alindima) [inline CREDITS.md:76]: kumargu@amazon.com is a duplicate, can be fixed also in .mailmap

File: src/firecracker/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "firecracker"
-version = "0.23.0"
+version = "0.24.0"
```
Comment (alindima) [inline src/firecracker/Cargo.toml:3]: Please also run a `cargo check`, so that we also update the lockfiles

Comment (sandreim) [reply src/firecracker/Cargo.toml:3]: Yeah, actually I missed to include cargo.lock in any of the commits.

Comment (sandreim) [reply src/firecracker/Cargo.toml:3]: Fixed.

File: CREDITS.md (hunk 2 above)
Comment (sandreim) [reply CREDITS.md:76]: Fixed.

**Comment by sandreim** [comment]
@AlexandruCihodaru @alindima PTAL.

**Review by serban300 — APPROVED** [review]

**Review by AlexandruCihodaru — APPROVED** [review]

File: CREDITS.md (hunk 1 above)
Comment (sandreim) [reply CREDITS.md]: Fixed.

**Pull request merged**